samples/hello-world/HW-pasqal-pulser.ipynb

```diff
@@ -0,0 +1,359 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 👋🌍 Hello, world: Submit a Pulser job to Pasqal\n",
+    "\n",
+    "In this notebook, we'll review the basics of Azure Quantum by submitting a simple *job*, or quantum program, to [Pasqal](https://pasqal.com/). We will use [Pulser](https://pulser.readthedocs.io/) to express the quantum job in an analog way."
+   ],
+   "id": "8493267a95ed8cbd"
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "source": [
+    "## Submit a simple job to Pasqal using Azure Quantum\n",
+    "Azure Quantum provides several ways to express quantum programs. In this example we are using Pulser, the analog quantum computing library of Pasqal. All code in this example will be written in Python.\n",
+    "\n",
+    "Let's begin. When you see a code block, hover over it and click the triangle play-button to execute it. To avoid any compilation issues, this should be done in order from top to bottom."
+   ],
+   "id": "808523d9ed21b01b"
+  },
+  {
+   "metadata": {},
+   "cell_type": "code",
+   "outputs": [],
+   "execution_count": null,
+   "source": [
+    "try:\n",
+    "    import pulser\n",
+    "except ImportError:\n",
+    "    !pip -q install pulser-core\n",
+    "    print(\"Please restart kernel to complete the installation of pulser.\")"
+   ],
+   "id": "261dab3c8880aeee"
+  },
```

Comment:
Since Pulser is a framework to construct quantum circuits (a kin to `Q#`, `Qiskit` and `Cirq`) I think we could:
1. Add another requirement to [azure-quantum](https://github.com/microsoft/azure-quantum-python/tree/main/azure-quantum), say `requirements-pulser.txt` with the following content:  
   ```
   pulser-core>=0.18,<1.0
   ```
   (assuming Pulser follows [SemVer](https://semver.org/))
The optional `requirements-pulser.txt` will get installed when user does:
   ```bash
   pip install azure-quantum[pulser]
   ```
   and it will get installed automatically when user does:
   ```bash
   pip install azure-quantum[all]
   ```
1. Remove this Notebooks cell, as the Notebooks in Azure Quantum will have all requirements installed by default.

FYI, @04diiguyi

Reply:
Yes Pulser follows SemVer, unfortunately after digging on this subject, I find that [v < 1.0 still allows breaking changes](https://github.com/pasqal-io/Pulser/blob/develop/release.md#versioning):

> Currently (as of July 2022), Pulser is still in an early development phase and has a MAJOR of 0 - as such, both breaking and backwards compatible changes and additions to the API will be introduced through increments in the MINOR until version 1.0.0 is released.

In that light, it seems safest to specify `pulser-core=0.18`.

Other than that, happy to apply your suggestion, should I do it now, or do we need to wait for someone else's approval ?

Reply:
@ArnaudPel the `pulser-core==0.18` sounds good. We could also use `pulser-core==0.18.*` if we're certain no breaking change will be introduced with a patch.
Could you please make the aforementioned changes? Please, let us know if any help needed.

Reply:
I have added the file `requirements-pulser.txt` in this PR, and consequently removed this cell.

I did not try to test that locally, is there a way to do so before this PR is merged ?

Reply:
Sure, you could:
1. Create a new Python environment (Conda, venv), 
2. Navigate to `azure-quantum` folder
3. Run 
```python
pip install -e .[pulser]
```
4. Run code from the notebook you put together

Reply:
Ah yes, after a bit of head scratching I figured out that I needed to wrap `.[pulser]` around quotes, then it worked like a charm :relieved: 

So I confirm that this approach seems to work locally.